contracts/ERC-721/PeripheralERC721Pool.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.8.4;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "./IPeripheralERC721Pool.sol";
+import "./IERC721Pool.sol";
+
+/// @title PeripheralERC721Pool
+/// @author Hifi
+contract PeripheralERC721Pool is IPeripheralERC721Pool {
+    /// PUBLIC NON-CONSTANT FUNCTIONS ///
+
+    /// @inheritdoc IPeripheralERC721Pool
+    function bulkDeposit(IERC721Pool pool, uint256[] calldata ids) external override {
+        // Checks: ids length must be greater than zero
+        if (ids.length == 0) {
+            revert PeripheralERC721Pool__InsufficientIn();
+        }
+
+        IERC721 erc721Asset = IERC721(pool.asset());
+
+        // `msg.sender` is the owner of the NFTs who will receive the pool tokens.
+        // Checks: The caller must have allowed this contract to transfer the NFTs.
+        if (!erc721Asset.isApprovedForAll(msg.sender, address(this))) revert PeripheralERC721Pool__UnapprovedOperator();
```

Comment:
This seems like a strong and unnecessary requirement ? Why not just let the code revert on `transferFrom` if there was no approval for the given tokens ? Right now if I own `N` >> 2 tokens, but want to `bulkDeposit` only 2 I must approve all `N` tokens.

Reply:
Thank you for your insightful review. your point is is indeed valid. 
I've taken your feedback into account and made the necessary changes in contract and foundry test in this [commit](https://github.com/hifi-finance/pooled-nft/pull/9/commits/5185c0b8a1e5b6bcd9dfc5b49b036ebf02aad1a8). 
Please let me know if there's anything else that you think could be improved.